bazel/spawn_proto_library.bzl

```diff
@@ -0,0 +1,27 @@
+"""
+This module fetches the spawn.proto file from the bazel source code and turns it into a proto_library target.
+"""
+
+def _spawn_proto_library_impl(ctx):
+    ctx.download(
+        url = "https://raw.githubusercontent.com/bazelbuild/bazel/refs/tags/{bazel_version}/src/main/protobuf/spawn.proto".format(bazel_version = native.bazel_version),
+        output = "spawn.proto",
+        sha256 = "381bcb109e2855d4055fdc75988cc2144e0a92aa8586298123bb662cdefa6afe",
```

Comment:
this will need to be changed whenever we update bazel, right? if the format is forwards compatible, would it make sense to stick to a (potentially outdated) version?

Reply:
Yeah why not. I stick to the spawn.proto from 7.6.0 and added a comment.

Reply:
I do see in [the git history of spawn.proto](https://github.com/bazelbuild/bazel/commits/master/src/main/protobuf/spawn.proto) that they made [a recent change](https://github.com/bazelbuild/bazel/commit/fd6a9833ba9fe18799193295b82134bc64ab17b3#diff-271d13b9a6de1e68cb8824ec12cf01aee40e4cd3d8339cea90b63e6bd3e9f539R237). But it doesn't seem to be getting a lot of updates so let's just change it to the spawn.proto from bazel-8 once we upgrade and hope it stays stable after that.